src/main/java/org/opentripplanner/routing/impl/PathFinder/PathFinderPolicy.java

```diff
@@ -0,0 +1,35 @@
+package org.opentripplanner.routing.impl.PathFinder;
+
+import org.opentripplanner.common.model.GenericLocation;
+import org.opentripplanner.routing.core.RoutingRequest;
+import org.opentripplanner.standalone.Router;
+
+public class PathFinderPolicy {
+
+    public static PathFinder selectPathFinder(RoutingRequest request, Router router) {
+        if(request.hasBikeLocation()) {
+            double distanceToOrigin = distance(request.from, request.bikeLocation);
+            double distanceToDestination = distance(request.bikeLocation, request.to);
+            if (request.modes.getBicycle() && distanceToOrigin < 500) {
+                return new WalkToBikePathFinder(router);
+            } else if (request.modes.getBicycle() && distanceToDestination < 3000) {
```

Comment:
This is the policy of the strategy pattern. Should get a comment explaining the distance logic :)